src/connectDialog.cpp

```diff
@@ -50,6 +51,10 @@ ConnectDialog::ConnectDialog(QWidget *widget) : QWidget(widget),
 	setDynamicProperty(ui->hostname, "invalid", false);
 	setDynamicProperty(ui->hostname, "valid", false);
 	ui->infoSection->hide();
+
+	ui->demoDevicesComboBox->addItem("ADALM2000");
+	ui->demoDevicesComboBox->setDisabled(true);
+	process = std::unique_ptr<QProcess>(new QProcess(this));
```

Comment:
Does passing `this` as parent to the `QProcess` not ensure that the object will be destroyed when `this` gets destroyed? If yes then having it stored also in unique_ptr would result in a double delete. Either remove `this` as parent or the unique_ptr.

Reply:
Modified.